### ISSUE TITLE
Sans priv, /add_to_tournament désinforme

### ISSUE THREAD

**Issue by mgl512** [Mynetest/pvpplus#1]
La commande répond qu'il n'y a pas de tournoi en cours quand on essaie d'ajouter quelqu'un au tournoi sans le priv de modérateur.

**Comment by paly2** [comment]
Bizarre, j'obtiens `You don't have permission to run this command (missing privilege: tournament_mod).`... Tu es sûr d'avoir exécuté la commande sans le privilège ?

**Comment by mgl512** [comment]
Oui, en deux occasions différentes alors qu'il y avait un tournoi en annonce sur le serveur.

**Comment by paly2** [comment]
Tu as le priv tournament_mod, donc tu peux utiliser la commande. Un tournoi en annonce n'est pas considéré par le mod comme un tournoi en cours.

**Comment by mgl512** [comment]
Je ne l'avais pas quand j'ai utilisé cette commande. Je me suis donné le priv après mon deuxième test. Le temps de faire ça le tournoi avait commencé et j'ai pu y ajouter ce joueur random.

**Comment by paly2** [comment]
Tu obtiens tous les privilèges automatiquement, je crois :-)

**Comment by mgl512** [comment]
Non, il faut que je me les donne explicitement. Je ne me suis pas donné tous ceux de serveressentiels, comme heal ou spawn. J'ai du regarder la liste des privilèges possibles pour trouver celui du tournoi, et me le donner.

**Comment by paly2** [comment]
> Je ne me suis pas donné tous ceux de serveressentiels, comme heal ou spawn.

Vérifie que tu ne les as pas obtenus automatiquement en faisant la liste de tes privilèges avec `/privs` 😃 

> J'ai du regarder la liste des privilèges possibles pour trouver celui du tournoi, et me le donner.

Il y a une commande pour faire ça ? Note: `/privs` sans argument retourne tes propres privilèges

**Comment by mgl512** [comment]
`/help privs ` donne la liste de tous les privilèges possibles je pense

**Comment by paly2** [comment]
> `/help privs` donne la liste de tous les privilèges possibles je pense

En effet, super, je l'ignorais !

Pour ce qui est du problème en lui-même, mes tests me donnent bien le message d'erreur attendu (effectués sur mon compte joueur `Upsilon2`, car je ne peux pas me retirer de privs sur `Upsilon`)...

**Comment by mgl512** [comment]
En jeu: /add_to_tournament dit qu'il n'y a pas de tournoi tant qu'il n'a pas réellement commencé. Le tournoi ne commence pas réellement, et /add_to_tournament répond qu'il n'y a pas de tournoi, tant qu'il n'y a pas au moins 2 joueurs déjà inscrits. 

matthieu a démarré un tournoi, je me suis engagé dedans et j'ai pu ajouter des joueurs de force après qu'il a commencé réellement. Ils ont quitté, matthieu aussi et le serveur a crashé quand j'ai tapé /leave_tournament, j'étais le dernier participant. 
```
2017-03-25 11:09:56: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod 'pvpplus' in callback on_chat_message(): /home/minetest/.minetest/mods/pvpplus/functions.lua:23: attempt to index local 'player' (a nil value)
2017-03-25 11:09:56: ERROR[Main]: stack traceback:
2017-03-25 11:09:56: ERROR[Main]: 	/home/minetest/.minetest/mods/pvpplus/functions.lua:23: in function 'pvp_set'
2017-03-25 11:09:56: ERROR[Main]: 	/home/minetest/.minetest/mods/pvpplus/tournament.lua:152: in function 'stop_tournament'
2017-03-25 11:09:56: ERROR[Main]: 	/home/minetest/.minetest/mods/pvpplus/tournament.lua:261: in function 'remove_from_tournament'
2017-03-25 11:09:56: ERROR[Main]: 	/home/minetest/.minetest/mods/pvpplus/tournament.lua:377: in function 'func'
2017-03-25 11:09:56: ERROR[Main]: 	...e/minetest/minetest/bin/../builtin/game/chatcommands.lua:48: in function <...e/minetest/minetest/bin/../builtin/game/chatcommands.lua:36>
2017-03-25 11:09:56: ERROR[Main]: 	/home/minetest/minetest/bin/../builtin/game/register.lua:412: in function </home/minetest/minetest/bin/../builtin/game/register.lua:392>
```

**Comment by paly2** [comment]
À retester, de nombreuses modifications ont eu lieu dans ce fichier (`functions.lua`) depuis.

**Comment by paly2** [comment]
@mgl512 Le problème existe-t-il toujours ?

**Comment by mgl512** [comment]
> Le problème existe-t-il toujours ?

Je n'ai pas vu de tournoi normal depuis longtemps, juste des spammeurs qui les enchainent sans but. Ils ne voient meme pas qu'on leur dit d'arrêter avec tout le texte qui s'affiche à l'écran.
